astropy/table/tests/test_groups.py

```diff
@@ -658,7 +653,7 @@ def all_positive(column):
         format="ascii",
     )
     tg = t.group_by("a")
-    c2 = tg["c"].groups.filter(all_positive)
+    c2 = tg["c"].groups.filter(lambda column: np.all(column >= 0))
```

Comment:
Didn't Python or some style check discourage `lambda` usage at some point? Is that not the case anymore?

Reply:
Not as far as I know. Maybe I just joined the game too late to hear about it 🤷🏻‍♂️

Reply:
What is discouraged (by both [PEP 8](https://peps.python.org/pep-0008/#programming-recommendations) and [Ruff rule E731 (lambda-assignment)](https://docs.astral.sh/ruff/rules/lambda-assignment/)) is assigning a `lambda` to a variable.

Reply:
Maybe ok here since we're not assigning it to anything. I might just have some vague memory from old over-eager style checks.